### PR TITLE
Fix failing SetProcessorTests.testCopyFromOtherField test

### LINKED ISSUE #69876: Failure in SetProcessorTests.testCopyFromOtherField
**Repro line**: 

`./gradlew ':modules:ingest-common:test' --tests "org.elasticsearch.ingest.common.SetProcessorTests.testCopyFromOtherField" -Dtests.seed=11E310B562CFC8CD`

**Reproduces locally?**: Yes

**Applicable branches**: master, 7.x, 7.11, 7.12

### PR DESCRIPTION
**Pull request by danhermann**
Fixes two problems with the `testCopyFromOtherField` test -- it should not choose a sub-object of the copy_from field as the target field and it should "deep-compare" any copied objects contained within maps as they will _not_ be reference-equal after the fix in #69349.

Fixes #69876.

### CODE REVIEW

File: modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/SetProcessorTests.java [hunk 1]
```diff
@@ -135,19 +136,47 @@ public void testSetMetadataIfPrimaryTerm() throws Exception {
...
+            Map<?, ?> actualMap = (Map<?, ?>) actual;
+            assertThat(actualMap.keySet().toArray(), arrayContainingInAnyOrder(expectedMap.keySet().toArray()));
+            for (Map.Entry<?, ?> entry : actualMap.entrySet()) {
+                if (entry.getValue() instanceof Map) {
+                    assertMapEquals(entry.getValue(), expectedMap.get(entry.getKey()));
+                } else {
+                    assertThat(entry.getValue(), equalTo(expectedMap.get(entry.getKey())));
+                }
+            }
+        }
```
Comment (danhermann) [inline modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/SetProcessorTests.java:179]: I would be happy if there were some Hamcrest magic that did this "deep-compare" of objects within maps automatically, but I couldn't find anything along those lines.

**Comment by danhermann** [comment]
@elasticmachine update branch

**Review by dakrone — APPROVED** [review]
LGTM

**Comment by danhermann** [comment]
Thanks, @dakrone!

**Pull request merged**